### PR TITLE
Move testToUtc test to DateFormattersTests

### LINKED ISSUE #38359: [CI] NodeStatsMonitoringDocTests#testToUTC fails
Build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/1812/console

Couldn't reproduce locally but looks like a date formatting issue.
```
./gradlew :x-pack:plugin:monitoring:unitTest \
  -Dtests.seed=1B6F728E6033F9E8 \
  -Dtests.class=org.elasticsearch.xpack.monitoring.collector.node.NodeStatsMonitoringDocTests \
  -Dtests.method="testToUTC" \
  -Dtests.security.manager=true \
  -Dtests.locale=lv-LV \
  -Dtests.timezone=GMT0 \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

```
org.junit.ComparisonFailure: expected:<2019-02-04T16:31:46[]Z> but was:<2019-02-04T16:31:46[.000]Z>
	at __randomizedtesting.SeedInfo.seed([1B6F728E6033F9E8:F583DD13D392E2A0]:0)
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.xpack.monitoring.exporter.BaseMonitoringDocTestCase.testToUTC(BaseMonitoringDocTestCase.java:173)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
```

### PR DESCRIPTION
**Pull request by pgomulka**
The test was relying on toString in ZonedDateTime which is different to
what is formatted by strict_date_time when milliseconds are 0
The method is just delegating to dateFormatter, so that scenario should
be covered there.

closes #38359

### CODE REVIEW

**Review by cbuescher — APPROVED** [review]
LGTM

File: server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java [hunk 1]
```diff
@@ -248,4 +252,15 @@ public void testRoundupFormatterLocale() {
         assertThat(roundupParser.getLocale(), is(locale));
         assertThat(formatter.locale(), is(locale));
     }
+
+    public void test0MillisAreFormatted() {
+        DateFormatter formatter = DateFormatter.forPattern("strict_date_time");
+
+        Clock clock = Clock.fixed(ZonedDateTime.of(2019,02,8,11,43,00,0,
+            ZoneOffset.UTC).toInstant(),ZoneOffset.UTC);
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java]: nit: add some spaces after the commas

File: server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java [hunk 2]
```diff
@@ -248,4 +252,15 @@ public void testRoundupFormatterLocale() {
...
+
+        String formatted = formatter.formatMillis(clock.millis());
+
+        assertThat(formatted, is("2019-02-08T11:43:00.000Z"));
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java]: nit: maybe ommit all the blank lines. Its a matter of taste but I tend to only use them in large methods to group semantically related blocks only.

**Comment by jpountz** [comment]
@pgomulka  I'm assuming there is nothing left to backport and removed the backport pending label.

**Pull request merged**